recipes/imagl/all/conanfile.py

```diff
@@ -0,0 +1,66 @@
+from conans import ConanFile, CMake, tools, errors
+
+class ImaglConan(ConanFile):
+    name = "imagl"
+    license = "GPL-3"
+    homepage = "https://github.com/Woazim/imaGL"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A lightweight library to load image for OpenGL application. Directly compatible with glTexImage* functions. Tested on linux, macOS and Windows with gcc, clang and msvc."
+    topics = ("opengl", "texture", "image")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake"
+    requires = [("libpng/1.6.37")]
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        # This small hack might be useful to guarantee proper /MT /MD linkage
+        # in MSVC if the packaged project doesn't have variables to set it
+        # properly
+        tools.replace_in_file(self.name + "-v" + self.version + "/CMakeLists.txt", "# conan insert",
+                              '''include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()''')
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.verbose = True
+        cmake.configure(source_folder=self.name + "-v" + self.version, defs={"STATIC_LIB": "OFF" if self.options.shared else "ON"})
+        cmake.build()
+        cmake.install()
+
+        # Explicit way:
+        # self.run('cmake %s/hello %s'
+        #          % (self.source_folder, cmake.command_line))
+        # self.run("cmake --build . %s" % cmake.build_config)
+
+    def package(self):
+        self.copy("*.h", dst="include", src=self.package_folder + "include")
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+        self.copy(self.name + "-v" + self.version + "/LICENSE", dst="licenses", keep_path=False)
+
+    def package_info(self):
+        debug_suffix = ""
+        static_suffix = ""
+        if self.settings.build_type == "Debug": debug_suffix = "d"
+        if not(self.options.shared): static_suffix = "s"
+        self.cpp_info.libs = ["imaGL{debug}{static}".format(debug = debug_suffix, static = static_suffix)]
+
+    def configure(self):
```

Comment:
This should be higher up in the order see this https://github.com/conan-io/conan-center-index/blob/ddd8c3c7499678583061a570b03cac615d67b157/docs/reviewing.md#order-of-methods-and-attributes

For managing the c++ version requirements... I'd suggest you copy this snippet
https://github.com/conan-io/conan/issues/8002